receiver/prometheusreceiver/internal/ocastore_test.go

```diff
@@ -0,0 +1,312 @@
+					numLbKeys: 0,
+					numTs:     1,
+				},
+				{
+					mname:     "test_my_summary_vec",
+					mtype:     metricspb.MetricDescriptor_SUMMARY,
+					numLbKeys: 2,
+					numTs:     3,
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			srv := startMockServer(tt.page)
+			target, _ := url.Parse(srv.URL)
+			defer srv.Close()
+			mc, cancel := startScraper(target.Host)
+			defer cancel()
+
+			var d *data.MetricsData
+			select {
+			case d = <-mc.Metrics:
+			case <-time.After(time.Second * 10):
+				t.Error("no data come back in 10 second")
+			}
+
+			if len(tt.mv) != len(d.Metrics) {
+				t.Errorf("Number of metrics got=%v exepct=%v\n", len(d.Metrics), len(tt.mv))
+			}
+
+			for i, dd := range d.Metrics {
+				got := v{
+					mname:     dd.MetricDescriptor.Name,
+					mtype:     dd.MetricDescriptor.Type,
+					numLbKeys: len(dd.MetricDescriptor.LabelKeys),
+					numTs:     len(dd.Timeseries),
+				}
+				if !reflect.DeepEqual(got, tt.mv[i]) {
+					t.Errorf("got %v, expect %v", got, tt.mv[i])
+				}
+			}
+		})
+	}
+
+}
+
+func startMockServer(page string) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		rw.Write([]byte(page))
+	}))
+}
+
+func startScraper(target string) (*mockConsumer, context.CancelFunc) {
+	rawCfg := fmt.Sprintf(scrapCfgTpl, target)
+
+	cfg, err := config.Load(rawCfg)
+	if err != nil {
+		fmt.Println(err)
+		return nil, nil
+	}
+	con := newMockConsumer()
+	o := NewOcaStore(context.Background(), con, testLogger)
+	scrapeManager := scrape.NewManager(log.NewNopLogger(), o)
+	o.SetScrapeManager(scrapeManager)
+
+	ctx := context.Background()
+	ctxScrape, cancelScrape := context.WithCancel(ctx)
+
+	discoveryManagerScrape := discovery.NewManager(ctxScrape, NewZapToGokitLogAdapter(zapLogger))
+	go discoveryManagerScrape.Run()
+	scrapeManager.ApplyConfig(cfg)
+
+	// Run the scrape manager.
+	syncConfig := make(chan bool)
+	errsChan := make(chan error, 1)
+	go func() {
+		defer close(errsChan)
+
+		<-time.After(100 * time.Millisecond)
+		close(syncConfig)
+		if err := scrapeManager.Run(discoveryManagerScrape.SyncCh()); err != nil {
+			errsChan <- err
+		}
+	}()
+	<-syncConfig
+	// By this point we've given time to the scrape manager
+	// to start applying its original configuration.
+
+	discoveryCfg := make(map[string]sd_config.ServiceDiscoveryConfig)
+	for _, scrapeConfig := range cfg.ScrapeConfigs {
+		discoveryCfg[scrapeConfig.JobName] = scrapeConfig.ServiceDiscoveryConfig
+	}
+
+	// Now trigger the discovery notification to the scrape manager.
+	discoveryManagerScrape.ApplyConfig(discoveryCfg)
+
+	return con, cancelScrape
+}
+
+var scrapCfgTpl = `
+
+scrape_configs:
+- job_name: 'test'
+  scrape_interval: 80ms
+  static_configs:
+  - targets: ['%s']
+`
+
+// https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md#text-format-example
+var testData1 = `
```

Comment:
Consider splitting the different metric types into different tests and add any missing metric types (e.g. gauge).

Also consider adding more edge cases (e.g. histogram with 1 bucket).